### PR TITLE
Update about_Foreach.md

### PR DESCRIPTION
**Pull request by sassdawe**
The last example was not working when the parameter was used and not a pipeline to call the function.
Please see the original and the updated version with pester tests in this folder [about_foreach](https://github.com/sassdawe/PowerShell/tree/master/from-docs/about_foreach)

I also added some **write-verbose** to the script to help explain a little bit more what's happening.

### CODE REVIEW

**Comment by zjalexander** [comment]
OPS Build status updates of commit _[189227c](https://github.com/sassdawe/PowerShell-Docs/pull/4002/commits/189227cfa89c65cf7fb2b8dc8af014088c09b029)_: 

### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[reference/5.1/Microsoft.PowerShell.Core/About/about_Foreach.md](https://github.com/sassdawe/PowerShell-Docs/blob/patch-1/reference/5.1/Microsoft.PowerShell.Core/About/about_Foreach.md) | :warning:Warning | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.core/about/about_foreach?view=powershell-5.1&branch=pr-en-us-4002) (powershell-5.1) | [Details](#user-content-reference%2F5.1%2FMicrosoft.PowerShell.Core%2FAbout%2Fabout_Foreach.md)

<a name="reference%2F5.1%2FMicrosoft.PowerShell.Core%2FAbout%2Fabout_Foreach.md"></a>
### [reference/5.1/Microsoft.PowerShell.Core/About/about_Foreach.md](https://github.com/sassdawe/PowerShell-Docs/blob/patch-1/reference/5.1/Microsoft.PowerShell.Core/About/about_Foreach.md)
 - **[Warning]** Metadata with following name(s) are not allowed to be set in Yaml header, or as file level metadata in docfx.json, or as global metadata in docfx.json: locale. They are generated by OPS, so the values set in these 3 places will be ignored. Please remove them from all 3 places to resolve the warning.

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C3%5C15%5C6ff7e8c3-dfc6-3ebd-da5a-d5e2ff43de8f%5CPullRequest%5C201903152200342807-4002%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=nI%2FZEqbONCN1jysO6IoYQuAKfoLplH6y3g9fcWV2ztk%3D&st=2019-03-15T21%3A59%3A53Z&se=2019-04-15T22%3A04%3A53Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Foreach.md [hunk 1]
```diff
@@ -175,19 +175,23 @@ function Get-FunctionPosition {
   process {
     try {
       $filesToProcess = if ($_ -is [System.IO.FileSystemInfo]) {
+        Write-Verbose "From pipeline"
         $_
       } else {
-        $filesToProcess = Get-Item -Path $Path
+        Write-Verbose "From parameter, $Path"
+        Get-Item -Path $Path
```
Comment (sdwheeler) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Foreach.md]: Why did you remove the assignment here? If this code path is used then the collection in the foreach loop is empty.

Comment (sassdawe) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Foreach.md]: Because the **if** starts with `$filesToProcess = if (...` therefor inside the **else** script block you cannot assign value to the same variable.
I tested it and didn't work.

Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Foreach.md]: DOH, I see it now. Thanks.

**Comment by zjalexander** [comment]
OPS Build status updates of commit _[036752c](https://github.com/sassdawe/PowerShell-Docs/pull/4002/commits/036752c5a0313cccb75c0255a26a5ed7b8ec4d49)_: 

### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[reference/5.1/Microsoft.PowerShell.Core/About/about_Foreach.md](https://github.com/sassdawe/PowerShell-Docs/blob/patch-1/reference/5.1/Microsoft.PowerShell.Core/About/about_Foreach.md) | :warning:Warning | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.core/about/about_foreach?view=powershell-5.1&branch=pr-en-us-4002) (powershell-5.1) | [Details](#user-content-reference%2F5.1%2FMicrosoft.PowerShell.Core%2FAbout%2Fabout_Foreach.md)

<a name="reference%2F5.1%2FMicrosoft.PowerShell.Core%2FAbout%2Fabout_Foreach.md"></a>
### [reference/5.1/Microsoft.PowerShell.Core/About/about_Foreach.md](https://github.com/sassdawe/PowerShell-Docs/blob/patch-1/reference/5.1/Microsoft.PowerShell.Core/About/about_Foreach.md)
 - **[Warning]** Metadata with following name(s) are not allowed to be set in Yaml header, or as file level metadata in docfx.json, or as global metadata in docfx.json: locale. They are generated by OPS, so the values set in these 3 places will be ignored. Please remove them from all 3 places to resolve the warning.

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C3%5C18%5C6ff7e8c3-dfc6-3ebd-da5a-d5e2ff43de8f%5CPullRequest%5C201903181802172658-4002%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=q9x4r86rRokDxbuOEfpeOYZJEtnOWioBsU3C65CsWj4%3D&st=2019-03-18T18%3A02%3A09Z&se=2019-04-18T18%3A07%3A09Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.
